generator/src/main/java/com/graphicsfuzz/generator/knownvaluegeneration/ExpressionGenerator.java

```diff
@@ -163,6 +168,45 @@ public Expr generateExpr(FactManager factManager,
     return result;
   }
 
+  private Expr generateUniformValue(Value value) {
+    if (value.valueIsUnknown()) {
+      return null;
+    }
+    if (!(value instanceof NumericValue)) {
```

Comment:
Restricting to 'numeric' means that we exclude 'vec' and 'mat' types.

I suggest you check that it's either numeric, or composite with a BasicType as its type.  Then have a special case to exclude matrices as per the issue below.